id/__main__.py

```diff
@@ -66,9 +73,16 @@ def main() -> None:
 
     logger.debug(f"parsed arguments {args}")
 
-    from . import detect_credential
+    from . import decode_oidc_token, detect_credential
 
-    print(detect_credential(args.audience))
+    token = detect_credential(args.audience)
+    if token and args.decode:
+        header, payload, signature = decode_oidc_token(token)
+        print(json.dumps(header, indent=4))
+        print(json.dumps(payload, indent=4))
+        print(signature)
```

Comment:
On the fence about whether this should print the signature or not. Probably not immediately useful and potential a source of leaking a valid token (e.g. if someone ran this in GitHub Actions, I'm not sure if it would mask everything here).

Reply:
+1 to it not being immediately useful -- the signature is only valid for a single serialization because of JSON's malleability, so printing it probably wouldn't guarantee that the user can verify it anyways 🙂